src/Servers/IIS/IIS/test/IIS.Tests/MaxRequestBodySizeTests.cs

```diff
@@ -0,0 +1,258 @@
+            {
+                using (var connection = testServer.CreateConnection())
+                {
+                    await connection.Send(
+                        "POST / HTTP/1.1",
+                        "Host: localhost",
+                        "Content-Length: " + payloadSize,
+                        "",
+                        payload);
+                    await connection.Receive(
+                        "HTTP/1.1 500 Internal Server Error");
+                }
+            }
+        }
+
+        [ConditionalFact]
+        public async Task RequestBodyTooLargeChunked()
+        {
+            var maxRequestSize = 0x1000;
+
+            BadHttpRequestException exception = null;
+            using (var testServer = await TestServer.Create(
+                async ctx =>
+                {
+                    try
+                    {
+                        while (true)
+                        {
+                            var num = await ctx.Request.Body.ReadAsync(new byte[2000]);
+                        }
+                    }
+                    catch (BadHttpRequestException ex)
+                    {
+                        exception = ex;
+                        throw ex;
+                    }
+                }, LoggerFactory, new IISServerOptions { MaxRequestBodySize = maxRequestSize }))
+            {
+                using (var connection = testServer.CreateConnection())
+                {
+                    await connection.Send(
+                        "POST / HTTP/1.1",
+                        "Transfer-Encoding: chunked",
+                        "Host: localhost",
+                        "",
+                        "1001",
+                        new string('a', 4097),
+                        "0",
+                        "");
+                    await connection.Receive("HTTP/1.1 413 Payload Too Large");
+                }
+            }
+
+            Assert.NotNull(exception);
+            Assert.Equal(CoreStrings.BadRequest_RequestBodyTooLarge, exception.Message);
+        }
+
+        [ConditionalFact]
+        public async Task EveryReadFailsWhenContentLengthHeaderExceedsGlobalLimit()
+        {
+            BadHttpRequestException requestRejectedEx1 = null;
+            BadHttpRequestException requestRejectedEx2 = null;
+            using (var testServer = await TestServer.Create(
+                async ctx =>
+                {
+                    var buffer = new byte[1];
+                    requestRejectedEx1 = await Assert.ThrowsAsync<BadHttpRequestException>(
+                        async () => await ctx.Request.Body.ReadAsync(buffer, 0, 1));
+                    requestRejectedEx2 = await Assert.ThrowsAsync<BadHttpRequestException>(
+                        async () => await ctx.Request.Body.ReadAsync(buffer, 0, 1));
+                    throw requestRejectedEx2;
+                }, LoggerFactory, new IISServerOptions { MaxRequestBodySize = 0 }))
```

Comment:
Do we verify that "Content-Length: 0" and empty chunked request bodies are still allowed with MaxRequestBodySize = 0?

Reply:
Added tests.